File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution 
{
public:
    int jump(vector<int>& nums) 
    {
        int maxindex = 0, count = 0, lastjump = 0;
        for (int i = 0; i < nums.size() - 1; i++) 
        {
            maxindex = max(maxindex, i + nums[i]); 
            if (i == lastjump) 
            {
                count++;
                lastjump = maxindex; 
                cout << count << " "; 
            }
        }
    
        return count;
    }
};

File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution(object):
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        farthest = 0
        current_jump_end = 0
        jumps = 0

        for i in range(len(nums) - 1):
    # Continuously update the absolute furthest index we could reach
            farthest = max(farthest, i + nums[i])
    
    # Once we reach the end of our current jump's territory...
            if i == current_jump_end:
                jumps += 1                  # We are forced to use a jump
                current_jump_end = farthest  # Our boundary extends to the best launcher we found
        return jumps